### PR TITLE
Replace func

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: expression/builtin/string.go [hunk 1]
```diff
@@ -146,3 +146,27 @@ func builtinUpper(args []interface{}, ctx map[interface{}]interface{}) (interfac
...
+
+// See: https://dev.mysql.com/doc/refman/5.0/en/string-functions.html#function_replace
+func builtinReplace(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	for _, arg := range args {
+		if arg == nil {
+			return nil, nil
+		}
+	}
+
+	str, err := types.ToString(args[0])
```
Comment (siddontang) [inline expression/builtin/string.go]: @shenli  ToString now handles DataItem or not?

File: expression/builtin/string.go [hunk 2]
```diff
@@ -146,3 +146,27 @@ func builtinUpper(args []interface{}, ctx map[interface{}]interface{}) (interfac
 		return strings.ToUpper(s), nil
 	}
 }
+
+// See: https://dev.mysql.com/doc/refman/5.0/en/string-functions.html#function_replace
+func builtinReplace(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	for _, arg := range args {
+		if arg == nil {
```
Comment (siddontang) [inline expression/builtin/string.go]: handle DataItem nil, ask @shenli  for more help.

File: expression/builtin/string.go [hunk 3]
```diff
@@ -146,3 +146,27 @@ func builtinUpper(args []interface{}, ctx map[interface{}]interface{}) (interfac
 		return strings.ToUpper(s), nil
 	}
 }
+
+// See: https://dev.mysql.com/doc/refman/5.0/en/string-functions.html#function_replace
```
Comment (qiuyesuifeng) [inline expression/builtin/string.go]: Use version 5.7 doc ref url.

File: expression/builtin/string.go (hunk 1 above)
Comment (shenli) [inline expression/builtin/string.go]: ToString handles DataItem now.

File: expression/builtin/string.go (hunk 2 above)
Comment (shenli) [inline expression/builtin/string.go]: Use types.IsNil() instead.

**Comment by zimulala** [comment]
PTAL

File: expression/builtin/builtin.go [hunk 4]
```diff
@@ -101,6 +101,7 @@ var Funcs = map[string]Func{
 	"lower":     {builtinLower, 1, 1, true, false},
 	"repeat":    {builtinRepeat, 2, 2, true, false},
 	"upper":     {builtinUpper, 1, 1, true, false},
+	"replace":   {builtinReplace, 3, 3, true, false},
```
Comment (qiuyesuifeng) [inline expression/builtin/builtin.go]: It is better to keep in order.

Comment (zimulala) [inline expression/builtin/builtin.go]: gotcha

**Comment by siddontang** [comment]
LGTM

**Pull request merged**